gtsam/gtsam.i

```diff
@@ -797,6 +797,12 @@ class Unit3 {
   size_t dim() const;
   gtsam::Unit3 retract(Vector v) const;
   Vector localCoordinates(const gtsam::Unit3& s) const;
+
```

Comment:
Thanks for the PR, @ayushbaid . Is Unit3 the only class that needs serialization here, and if so, why note Rot3, Pose3, etc?

Reply:
All other classes already have the `serialization` and `equals` API exposed in gtsam.i. `Unit3` was missing it.